### ISSUE TITLE
No IPv6 with dnsutils (bind-tools)

### ISSUE THREAD

**Issue by Phoenix1747** [HorlogeSkynet/archey4#19]
## Expected behavior
Sorry to spam you kind of but yeah, lol. Without dnsutils I get the following (somewhat) right output:
```
LAN_IP: 192.168.0.202
WAN_IP: <MY CORRECT IPv6>
```

## Current behavior
The problem is now that after I installed dnsutils the output just goes full ipv4:
```
LAN_IP: 192.168.0.202
WAN_IP: <MY CORRECT IPv4>
```

Maybe this is intended to be like that? I'm genuinely confused, because both times it returns the correct IPs, but only one at a time and it changes between ipv4 and 6.

EDIT: Bind-Tools version: 9.11.2.P1-1

**Comment by HorlogeSkynet** [comment]
It's okay, it increases the activity 🤣

--> If sometimes yout IPv6 output, I assume you didn't change the default configuration then.

In order to debug this behavior, I'm afraid you'll have to try the different commands in both cases unless we would not understand a thing...

```
dig +short -6 aaaa myip.opendns.com @resolver1.ipv6-sandbox.opendns.com
wget -qO- https://v6.ident.me/

dig +short myip.opendns.com @resolver1.opendns.com
wget -qO- https://v4.ident.me/
```

**Comment by Phoenix1747** [comment]
| command | with bind-tools | without |
| ------------- | ------------------ | --------- |
| dig +short -6 aaaa myip.opendns.com @resolver1.ipv6-sandbox.opendns.com | pass | no command |
| wget -qO- https://v6.ident.me/ | pass | pass |
| dig +short myip.opendns.com @resolver1.opendns.com | pass | no command |
| wget -qO- https://v4.ident.me/ | pass | pass |

**Comment by HorlogeSkynet** [comment]
Well I think we got lost 🤣

Two things :

1. On your system it's `bind-tools` that provides `dig` command ? I got an Arch Linux ARM, and I'm pretty sure it's `dnsutils` too...

2. The program is supposed to output the same stuff, with or without `dnsutils` (at least, that's how it has been thought recently with @Si13n7)

**Comment by Phoenix1747** [comment]
1. Bind-Tools is the package which provides dnsutils, there's no dedicated dnsutils package; [have a link](https://www.archlinux.org/packages/extra/x86_64/bind-tools/)
2. Yeah that would be nice, but as I described, with dns-utils/bind-tools/whatever it only gives the ipv4, without the correct ipv6.

**Comment by HorlogeSkynet** [comment]
1. Okay so I guess `dnsutils` was a particularity of ARM 🤔 🤔 We'll have to update the dependencies then...

2. You said the `dig` for IPv6 passed. So I don't really understand why the script does not output it in your case then 🤐

**Comment by Phoenix1747** [comment]
Just checked it again, manually using `dig` works, but the script only prints my IPv4 WAN 😕

**Comment by HorlogeSkynet** [comment]
Ahhh, I think I got that. Look at the `check_output` calls, I've set there `timeout=0.5` to avoid time loss during execution.
But if your queries are a bit "slow" on your network, you may experience this issue...

**Comment by Phoenix1747** [comment]
Just noticed that too this might be it.

**Comment by HorlogeSkynet** [comment]
Okay, try to replace `timeout=0.5` by `timeout=1`. If it works, we might consider increase the limit. What do you think about that ?

**Comment by Phoenix1747** [comment]
Oh no

**Comment by Phoenix1747** [comment]
The output now is like
```
LAN_IP  192.168.0.202, <MY CORRECT IPV6 WAN IP what the crap >
WAN_IP <MY CORRECT IPV4>, <MY CORRECT IPV6>
```
and sometimes it's just (I guess due to timeout) this:
```
LAN_IP  192.168.0.202, <MY CORRECT IPV6 WAN IP what the crap >
WAN_IP <MY CORRECT IPV4>
```

**Comment by HorlogeSkynet** [comment]
It definitely looks like timeout stuffs 😉

**Comment by Phoenix1747** [comment]
But why is my WAN IPv6 showing up at the LAN_IP? 😮

**Comment by HorlogeSkynet** [comment]
I don't think this is a mistake as it appears only if it matches `global` or `site` scopes...

**Comment by Phoenix1747** [comment]
Ok, get it. :+1:

**Comment by HorlogeSkynet** [comment]
But, again, maybe we're totally wrong here too 🤣
If you find a better way to do this, don't hesitate to point it out 👌

**Comment by Phoenix1747** [comment]
Sure :D

**Comment by AdJaGu** [comment]
@Phoenix1747 for me on this computer my LAN and WAN IPv6 address looks like `xxxx:xxxx:yyyy:yyyy::15` and on my other computer the LAN and WAN IPv6 address looks like `xxxx:xxxx:yyyy:yyyy::9`. The only difference comes after `::`. This computer being `15` and my other being `9`.

From my understanding each device that connects to my router gets its own IPv6 address. This address is the same on the LAN and WAN side. So, locally and globally this computer I am currently using is available at `xxxx:xxxx:yyyy:yyyy::15` and my other computer is available locally and globally at `xxxx:xxxx:yyyy:yyyy::9`.

IPv4 is different. There is not enough addresses to give every device its own IPv4 address. Each device that connects to my router is given an internal (LAN) IPv4 address. This computer is `192.168.1.50` and my other computer is `192.168.1.10`. But, these addresses only work locally. To reach any service on any of my computers I would need to use my external (WAN) IPv4 address. This external IPv4 address is completely different than my internal ones.

TLDR;
IPv4 addresses are limited. Because of this limit LAN IPv4 and WAN IPv4 are different.
IPv6 addresses for all the things! LAN IPv6 and WAN IPv6 are the same. Each computer/device has it's own IPv6 address (router too).

**Comment by Phoenix1747** [comment]
Ah, cool, thanks for the info @AdJaGu. The only question that's left to be cleared is why does bind-tools only show me my WAN IPv4 and without it does show both 🤔

**Comment by HorlogeSkynet** [comment]
Thanks here too, this was something I wasn't sure about 👍
So in this case, the `(global|site)` regex makes totally sense according to the IPv6 scopes 👌
